src/dmd/dsymbolsem.d

```diff
@@ -187,7 +187,7 @@ private extern (C++) FuncDeclaration buildPostBlit(StructDeclaration sd, Scope*
          * When the following field postblit calls fail,
          * this field should be destructed for Exception Safety.
          */
-        if (!sdv.dtor)
+        if (!sdv.dtor || (cast(TypeFunction)sdv.postblit.type).isnothrow)
```

Comment:
I think this should actually use the [`canThrow` function](https://github.com/dlang/dmd/blob/6371942b1bf26dc6c65cc7e0668bc82e1974c18d/src/dmd/canthrow.d#L36), but I can't figure out how to make a proper call to that function.  What should I pass for `Expression e`?